text/0162-refactoring-support.md

```diff
@@ -0,0 +1,677 @@
+  error.
+
+### Explicit mapping
+
+Although the CLI will automatically refactor only what is unambiguous, you may
+still need to have more control over the refactoring process. Companies usually
+have stringent policies on how changes are made to the production environment,
+for example. It could be that your company requires that every change must be
+explicitly declared in some sort of code, including refactors (as opposed to
+computed on-the-fly).
+
+In a situation like this, you can import and export mapping files. Here is how
+it works: at development time, you made a change that the CLI detected as a
+refactor. Since you want that refactor to be propagated to other environments,
+you export the mapping file, with the command
+`cdk refactor --export-mapping=file.json`.
+
+There are at least two possible paths from here, depending on your constraints:
+
+1. You send the exported file to an operations team, who will review it. If
+   approved, they manually run the command
+   `cdk refactor --import-mapping=file.json` on every protected environment in
+   advance (i.e., before your changes get deployed to those environments). When
+   you import a mapping, the CLI won't try to detect refactors.
+2. The `--apply-mapping` option is also available for the `deploy` command. So
+   you can commit the mapping file to version control, and configure your
+   pipeline to use it on every deployment. This is a more convenient option, as
+   it requires less coordination between different roles. Every time a refactor
+   is applied, a record of it is stored in the environment. This is to prevent
+   the same refactor from being applied multiple times.
+
+In general, if the protected environment is not in the same state as the
+environment where the mapping was generated, the `refactor --apply-mapping`
+command will fail.
+
+You can also use explicit mappings to define your own refactors, when the CLI
+didn't detect them automatically. This may happen a resource is moved and
+modified at the same time, for example.
+
+### Settings
+
+You have a few settings available to control the behavior of the refactoring
+feature.
+
+For both `deploy` and `refactor`:
+
+- `--export-mapping=<FILE>`: writes the mapping to a file. The file can be used
+  later to apply the same refactors in other environments.
+- `--import-mapping=<FILE>`: use the mapping from a file, instead of computing
+  it. This file can be generated using the `--export-mapping` option.
+- `--unstable=refactor`: used to acknowledge that this feature is experimental.
+  If the flag is not set, and the CLI would try to perform some refactor, the
+  command fails with an error message explaining why.
+
+For `deploy` only:
+
+- `--refactoring-action=[ACTION]`: the action to take in case there is a
+  refactor to be made. Possible values for `ACTION` are:
+    - `confirm`: ask the user what to do. This is the scenario described in the
```

Comment:
For `confirm`, how would we provide this programmatically? Or would it be a CLI-only option?